http_check/datadog_checks/http_check/http_check.py

```diff
@@ -98,71 +102,28 @@ def send_status_down(loginfo, down_msg):
-                            "Parameter disable_ssl_validation for {} is not explicitly set, "
-                            "defaults to true".format(addr)
-                        )
-
-            instance_proxy = self.get_instance_proxy(instance, addr)
-            self.log.debug("Proxies used for {} - {}".format(addr, instance_proxy))
-
-            auth = None
-            if password is not None:
-                if username is not None:
-                    auth = (username, password)
-                elif ntlm_domain is not None:
-                    auth = HttpNtlmAuth(ntlm_domain, password)
-
-            sess = requests.Session()
-            sess.trust_env = False
+            self.http.session.trust_env = False
             if weakcipher:
                 base_addr = '{uri.scheme}://{uri.netloc}/'.format(uri=parsed_uri)
-                sess.mount(base_addr, WeakCiphersAdapter())
+                self.http.session.mount(base_addr, WeakCiphersAdapter())
                 self.log.debug(
                     "Weak Ciphers will be used for {}. Supported Cipherlist: {}".format(
                         base_addr, WeakCiphersHTTPSConnection.SUPPORTED_CIPHERS
                     )
                 )
 
-            with warnings.catch_warnings():
-                # Suppress warnings from urllib3 only if disable_ssl_validation is explicitly set to True
-                #  and ignore_ssl_warning is True
-                if suppress_warning:
-                    warnings.simplefilter('ignore', InsecureRequestWarning)
-
-                # Add 'Content-Type' for non GET requests when they have not been specified in custom headers
-                if method.upper() in DATA_METHODS and not headers.get('Content-Type'):
-                    headers['Content-Type'] = 'application/x-www-form-urlencoded'
-
-                r = sess.request(
-                    method.upper(),
-                    addr,
-                    auth=auth,
-                    timeout=timeout,
-                    headers=headers,
-                    proxies=instance_proxy,
-                    allow_redirects=allow_redirects,
-                    stream=stream,
-                    verify=False if disable_ssl_validation else instance_ca_certs,
-                    json=data if method.upper() in DATA_METHODS and isinstance(data, dict) else None,
-                    data=data if method.upper() in DATA_METHODS and isinstance(data, string_types) else None,
-                    cert=(client_cert, client_key) if client_cert and client_key else None,
-                )
-
+            # Add 'Content-Type' for non GET requests when they have not been specified in custom headers
+            if method.upper() in DATA_METHODS and not headers.get('Content-Type'):
+                self.http.options['headers']['Content-Type'] = 'application/x-www-form-urlencoded'
+
+            r = getattr(self.http, method.lower())(
+                addr,
+                persist=True,
```

Comment:
It doesn't look like the `persist_connections` parameter from the config file is ever used then. I understand why we set persist to True here (so that we use the underlying session object instead of the request directly).

Should we remove that config flag ?